### ISSUE TITLE
TLSR8250

### ISSUE THREAD

**Issue by Hunter32R** [Marplex/tuyaFree#1]
Is it possible to run this firmware on a chip with 512Kb memory?

**Comment by Marplex** [comment]
Yes it's possible, but in BootLoader mode the firmware must be under 196k for 512K chips.
Just change `CHIP_TYPE` to `TLSR_8258_512K` here:
https://github.com/Marplex/tuyaFree/blob/32ef2a005538a86e89814b30ad324f5a1ab9b4f6/src/tuyaFree/version_cfg.h#L30C1-L30C35

**Comment by Hunter32R** [comment]
The firmware does not start. Maybe something else needs to be changed? I use the TB-03F module

**Comment by Marplex** [comment]
This repo is not just a firmware. I'm planning to turn this into an easy to use environment for building telink ZigBee firmwares.

The standard project `src/tuyaFree` doesn't do anything and it's meant to be used as a starter.

I'm planning to continue building this environment by developing a replacement firmware for the AliExpress ZigBee mini switches.
